sdk/src/main/java/com/horizen/account/secret/PrivateKeySecp256k1Creator.java

```diff
@@ -45,11 +39,7 @@ public PrivateKeySecp256k1 generateSecret(byte[] seed) {
     }
 
     @Override
-    public PrivateKeySecp256k1 generateNextSecret(NodeWalletBase wallet) {
-        List<Secret> prevSecrets = wallet.secretsOfType(PrivateKeySecp256k1.class);
-        byte[] nonce = Ints.toByteArray(prevSecrets.size());
-        byte[] seed = Keccak256.hash(Bytes.concat(wallet.walletSeed(), nonce));
-
-        return generateSecret(seed);
+    public byte[] salt() {
+        return "PrivateKeySecp25519k1".getBytes();
```

Comment:
Maybe is just a cosmetic issue but do you think it is better to move this variable at class level as final and not at the method level with something more generic? The same for the other private key classes

Reply:
Done

Reply:
- Being an immutable constant that has to be the same for every instance of the class it can also be put static and in uppercase. 
- Moreover, the UTF-8 charset specification for every `String.getBytes()` invocation need to be added as reported in the PR https://github.com/HorizenOfficial/Sidechains-SDK/pull/660